fiftyone/core/annotation/attributes.py

```
@@ -6,9 +6,15 @@
 |
 """
 
+import abc
 from dataclasses import dataclass
 from enum import Enum
-from typing import Any, Literal, Optional, Union
+from typing import Any, Generator, Literal, Optional, Union
```

Ruff (0.15.12)

[warning] Import from `collections.abc` instead: `Generator`

Import from `collections.abc`

(UP035)